pkg/nvpci/nvpci.go

```diff
@@ -281,6 +281,7 @@ func (p *nvpci) getGPUByPciBusID(address string, cache map[string]*NvidiaPCIDevi
 	}
 
 	driver, err := filepath.EvalSymlinks(path.Join(devicePath, "driver"))
+	// nolint:gocritic
```

Comment:
Instead of disabling linting (which should be `//nolint:gocritic` for machine-readable comments), should we factor this logic into a function instead:
```
func getDriver(devicePath string) (string, error) {
	driver, err := filepath.EvalSymlinks(path.Join(devicePath, "driver"))
	switch {
	case os.IsNotExist(err):
		return "", nil
	case err == nil:
		return filepath.Base(driver), nil
	}
	return "", err
}
```

This cleans up the callsite to:
```
	driver, err := getDriver(devicePath)
	if err != nil {
		return nil, fmt.Errorf("unable to detect driver for %s: %v", address, err)
	}
```

We could even introduce a type alias for a `string` that we could add this method and the methods below to.

Reply:
Sure

Reply:
Addressed the gocritic issues which were previously ignored. Please check now @elezar